pkg/slayers/path/mac.go

```diff
@@ -69,6 +86,14 @@ func FullMAC(h hash.Hash, info InfoField, hf HopField, buffer []byte) (res []byt
 	}
 	//@ assert h.Size() >= 16
 	res = h.Sum(buffer[:0])[:16]
+
+	// This is our "MAC assumption", linking the abstraction of the concrete 
+	// MAC tag to the abstract computation of the MAC tag.
```

Comment:
please move this comment to the line immediately before the call to `AssumeForIO`